app/models/proxy.rb

```diff
@@ -536,44 +521,76 @@ class PoliciesConfig
 
     validate :policies_configs_are_correct
 
+    DEFAULT_POLICY = { 'name' => 'apicast', 'humanName' => 'APIcast policy', 'description' => 'Main functionality of APIcast.',
+                       'configuration' => {}, 'version' => 'builtin', 'enabled' => true, 'removable' => false, 'id' => 'apicast-policy'  }.freeze
+
     def initialize(policies_config)
-      @policies_config = policies_config.map { |attrs| PolicyConfig.new(attrs) }
+      parsed_config = read_and_parse_policies_config(policies_config)
+
+      if parsed_config.detect { |c| c['name'] == DEFAULT_POLICY['name'] }
+        parsed_config
+      else
+        parsed_config.push(DEFAULT_POLICY)
+      end
+
+      @policies_config = parsed_config.map { |attrs| PolicyConfig.new(attrs) }
     end
 
+    # def policies_config=(attr_policies_config)
+    #   super(attr_policies_config.is_a?(String) ? attr_policies_config : attr_policies_config.to_json)
+    # end
+
     def self.name
       'PoliciesConfig'
     end
 
+    def to_json(*_args)
+      policies_config.map(&:to_h).to_json(*_args)
+    end
+
+    def chain
+      policies_config.select(&:enabled).map(&:to_h).stringify_keys.slice('name', 'version', 'configuration')
+    end
+
     private
 
+    def read_and_parse_policies_config(attr_policies_config)
+      read_and_parse_policies_config!(attr_policies_config)
+    rescue JSON::ParserError
+      logger.error "Invalid attribute policies_config: #{attr_policies_config}"
+      []
+    end
+
+    def read_and_parse_policies_config!(attr_policies_config)
+      attr_policies_config.blank? ? [] : Array(JSON.parse(attr_policies_config))
+    end
+
     def policies_configs_are_correct
       policies_config.each do |policy_config|
         # TODO: 5: errors.merge!(policy_config.errors)
         policy_config.errors.each { |attribute, message| errors.add(attribute, message) } unless policy_config.valid?
       end
+      unless policies_config.detect { |c| c.name == DEFAULT_POLICY['name'] }
+        errors.add(:policies_config, "must include default policy")
+      end
     end
   end
 
-  def read_and_parse_policies_config
-    read_and_parse_policies_config!
-  rescue JSON::ParserError
-    []
-  end
+  class PoliciesConfigType < ActiveRecord::Type::Text
```

Comment:
Yesterday we spoke about this. I believe we better extend `ActiveRecord` rather than `ActiveModel` because in Rails 2.5 they are the same class. But in 2.6 they are not. And here we are dealing with databases. So `ActiveRecord` appears to me as the appropriate option that should work with future Rails versions as well.